### PR TITLE
Merge doxygen2 into develop

### PR DESCRIPTION
**Pull request by gheber**
This PR contains the full (?) Doxygen-based RM. The current `doxygen-2` branch is no longer necessary (it's become an impediment) and all future changes should go straight to develop.

### CODE REVIEW

File: src/H5Ppublic.h [hunk 1]
```diff
@@ -1701,7 +1748,7 @@ H5_DLL herr_t H5Punregister(hid_t pclass_id, const char *name);
 /* Object creation property list (OCPL) routines */
 
 /**
- * \ingroup OCPL
+ * \ingroup DCPL
```
Comment (qkoziol) [inline src/H5Ppublic.h:1751]: OCPL was correct for this routine.

File: src/H5Ppublic.h [hunk 2]
```diff
@@ -1798,7 +1845,7 @@ H5_DLL herr_t H5Pget_attr_creation_order(hid_t plist_id, unsigned *crt_order_fla
  */
 H5_DLL herr_t H5Pget_attr_phase_change(hid_t plist_id, unsigned *max_compact, unsigned *min_dense);
 /**
- * \ingroup OCPL
+ * \ingroup DCPL
```
Comment (qkoziol) [inline src/H5Ppublic.h]: OCPL was correct for this routine.

File: src/H5Ppublic.h [hunk 3]
```diff
@@ -1865,7 +1912,7 @@ H5_DLL H5Z_filter_t H5Pget_filter2(hid_t plist_id, unsigned idx, unsigned int *f
                                    size_t *cd_nelmts /*out*/, unsigned cd_values[] /*out*/, size_t namelen,
                                    char name[], unsigned *filter_config /*out*/);
 /**
- * \ingroup OCPL
+ * \ingroup DCPL
```
Comment (qkoziol) [inline src/H5Ppublic.h]: OCPL was correct for this routine.

File: src/H5Ppublic.h [hunk 4]
```diff
@@ -1923,7 +1970,7 @@ H5_DLL herr_t H5Pget_filter_by_id2(hid_t plist_id, H5Z_filter_t filter_id, unsig
                                    size_t *cd_nelmts /*out*/, unsigned cd_values[] /*out*/, size_t namelen,
                                    char name[] /*out*/, unsigned *filter_config /*out*/);
 /**
- * \ingroup OCPL
+ * \ingroup DCPL
```
Comment (qkoziol) [inline src/H5Ppublic.h]: OCPL was correct for this routine.

File: src/H5Ppublic.h [hunk 5]
```diff
@@ -1983,7 +2030,7 @@ H5_DLL int H5Pget_nfilters(hid_t plist_id);
  */
 H5_DLL herr_t H5Pget_obj_track_times(hid_t plist_id, hbool_t *track_times);
 /**
- * \ingroup OCPL
+ * \ingroup DCPL
```
Comment (qkoziol) [inline src/H5Ppublic.h]: OCPL was correct for this routine.

File: src/H5Ppublic.h [hunk 6]
```diff
@@ -2012,7 +2059,7 @@ H5_DLL herr_t H5Pget_obj_track_times(hid_t plist_id, hbool_t *track_times);
 H5_DLL herr_t H5Pmodify_filter(hid_t plist_id, H5Z_filter_t filter, unsigned int flags, size_t cd_nelmts,
                                const unsigned int cd_values[/*cd_nelmts*/]);
 /**
- * \ingroup OCPL
+ * \ingroup DCPL
```
Comment (qkoziol) [inline src/H5Ppublic.h]: OCPL was correct for this routine.

File: src/H5Ppublic.h [hunk 7]
```diff
@@ -2166,7 +2213,7 @@ H5_DLL herr_t H5Pset_attr_creation_order(hid_t plist_id, unsigned crt_order_flag
  */
 H5_DLL herr_t H5Pset_attr_phase_change(hid_t plist_id, unsigned max_compact, unsigned min_dense);
 /**
- * \ingroup OCPL
+ * \ingroup DCPL GCPL
```
Comment (qkoziol) [inline src/H5Ppublic.h]: OCPL was correct for this routine.

File: src/H5Ppublic.h [hunk 8]
```diff
@@ -2222,7 +2269,7 @@ H5_DLL herr_t H5Pset_attr_phase_change(hid_t plist_id, unsigned max_compact, uns
  */
 H5_DLL herr_t H5Pset_deflate(hid_t plist_id, unsigned level);
 /**
- * \ingroup OCPL
+ * \ingroup DCPL GCPL
```
Comment (qkoziol) [inline src/H5Ppublic.h]: OCPL was correct for this routine.

File: src/H5Ppublic.h [hunk 9]
```diff
@@ -2487,7 +2534,7 @@ H5_DLL herr_t H5Pset_deflate(hid_t plist_id, unsigned level);
 H5_DLL herr_t H5Pset_filter(hid_t plist_id, H5Z_filter_t filter, unsigned int flags, size_t cd_nelmts,
                             const unsigned int c_values[]);
 /**
- * \ingroup OCPL
+ * \ingroup DCPL GCPL
```
Comment (qkoziol) [inline src/H5Ppublic.h]: OCPL was correct for this routine.

File: src/H5Ppublic.h (hunk 1 above)
Comment (gheber) [reply src/H5Ppublic.h:1751]: Just to be clear, the description of the `plist_id` parameter should then read:

```
 * \param[in] plist_id Object creation property list identifier
```

Right?

Comment (qkoziol) [reply src/H5Ppublic.h:1751]: > Just to be clear, the description of the `plist_id` parameter should then read:
> 
> ```
>  * \param[in] plist_id Object creation property list identifier
> ```
> 
> Right?

Yes

**Review by ChristopherHogan — APPROVED** [review]

**Review by epourmal — APPROVED** [review]
Please add a record to the RELEASE.txt file (Documentation section) about the new feature and how to enable it.
Thank you!

**Review by lrknox — APPROVED** [review]

**Pull request merged**